packages/amplify-cli/src/init-steps/s0-analyzeProject.ts

```diff
@@ -220,6 +230,108 @@ function getDefaultEnv(context: $TSContext): string | undefined {
   return undefined;
 }
 
+async function setEnvVar(context: $TSContext, currentEnvName: string, projectPath: string) {
+  const teamProviderInfo = stateManager.getTeamProviderInfo(projectPath, {
+    throwIfNotExist: false,
+    default: {},
+  });
+  const functions = teamProviderInfo[currentEnvName]?.categories?.function || {};
+  const functionNames = Object.keys(functions);
+  if (functionNames.length === 0) {
+    return {};
+  }
+
+  const envVarQuestion = async () => {
+    const envVarQuestion: inquirer.ListQuestion = {
+      type: 'list',
+      name: 'envVar',
+      message: 'You have custom function environment variables configured. What do you want to do?',
+      choices: [
+        {
+          value: 'carry',
+          name: 'Carry existing values over to the new environment (values can be updated later using amplify update function)',
+        },
+        {
+          value: 'update',
+          name: 'Update values now',
+        },
+      ],
+    };
+    if (context.parameters.options.quickstart) return;
+    const { envVar } = await inquirer.prompt(envVarQuestion);
+    if (envVar === 'carry') return;
+    if (envVar === 'update') await envVarSelectFunction();
+  };
+
+  const envVarSelectFunction = async () => {
+    const abortKey = uuid.v4();
+    const functionNameQuestion: inquirer.ListQuestion = {
+      type: 'list',
+      name: 'functionName',
+      message: 'Select the Lambda function you want to update values',
+      choices: functionNames
+        .map(name => ({
+          name,
+          value: name,
+        }))
+        .concat({
+          name: "I'm done",
+          value: abortKey,
+        }),
+    };
+    const { functionName } = await inquirer.prompt(functionNameQuestion);
+    if (functionName === abortKey) return;
+    await envVarSelectKey(functionName);
+  };
+
+  const envVarSelectKey = async (functionName: string) => {
+    const envVars = functions[functionName];
+    const abortKey = uuid.v4();
+    const keyNameQuestion: inquirer.ListQuestion = {
+      type: 'list',
+      name: 'keyName',
+      message: 'Which environment variable do you want to update:',
+      choices: Object.keys(envVars)
```

Comment:
we should list the actual env var name, not the CFN parameter name

Reply:
The choices are from `team-provider-info.json` instead of CFN parameter names and that's what we care to carry over.

Reply:
right, but the CFN parameter name is an implementation detail that is hidden from the customer. They should only know about the lambda function env var

Reply:
I replaced `team-provider-info` with the actual env vars! Thanks :)